src/Node/FS/Perms.purs

```diff
@@ -0,0 +1,86 @@
+module Node.FS.Perms
+  ( permsFromString
+  , permsToString
+  , permsToNum
+  , none
+  , r
+  , w
+  , x
+  , mkPerms
+  , Perms()
+  , Perm()
+  ) where
+
+import Global (readInt)
+import Data.Maybe (Maybe(..))
+import Data.Char (Char(), charString)
+import Data.String (toCharArray)
+import Data.Function
+
+data Perm = Perm { r :: Boolean, w :: Boolean, x :: Boolean }
+data Perms = Perms { u :: Perm, g :: Perm, o :: Perm }
```

Comment:
We can make `Perm` and `Perms` type aliases, or at least newtypes. Then we can avoid exporting `mkPerms` which has that nasty unnecessary ordering in its parameters.